### ISSUE TITLE
Add support for seccomp

### ISSUE THREAD

**Issue by vishh** [kubernetes/kubernetes#20870]
Starting from v1.10 docker daemon will run with a [restrictive seccomp profile](https://github.com/docker/docker/blob/master/docs/security/seccomp.md) turned on by default.

Users will have to include a custom seccomp profile in addition to capabilities. 

Until we figure out a sane way to add support for seccomp, I propose disabling (`unconfined`) seccomp profiles by default.
This will allow users to upgrade to docker v1.10 seamlessly.

cc @kubernetes/goog-node @bgrant0607

**Comment by vishh** [comment]
cc @erictune @kubernetes/rh-platform-management

**Comment by bgrant0607** [comment]
cc @pmorie @pweil-

**Comment by ncdc** [comment]
cc @kubernetes/rh-cluster-infra

**Comment by rhatdan** [comment]
Is the problem here that you are worried about seccomp breaking certain actions?  I hate to turn off a key security feature of docker  for the masses of users of k8s.

**Comment by vishh** [comment]
@rhatdan: https://github.com/docker/docker/issues/20082

**Comment by erictune** [comment]
Do these make sense as requirements for kubernetes support for seccomp filters:

1. we, the project, pre-defines names for certain "canned" seccomp filters with sensible defaults
  -  probably the same as Docker's 1.10 default, and unrestricted.
1. cluster admins can define their own custom filters too. 
1. it should be possible to define a seccomp filter and a list of allowed capabilities in the same policy object, so that both are set on a pod/container at once, to avoid user confusion about the need to set both.
1. users can select the policy that a pod/container uses by giving the short name of a policy object, rather than inlining the entire seccomp filter text into the pod spec.
1. admins can change the default policy for pods that don't say what to do as regards seccomp.
1. the default default, on a new/upgraded cluster does not change, without some affirmative user action.
1. if a user writes a pod spec that asks for a level of security that is not supported by the node (e.g. they ask for a restrictive seccomp profile, and the node is running docker 1.9) then pod creation fails, rather then the user silently getting no security.

**Comment by erictune** [comment]
@kubernetes/kube-iam

**Comment by erictune** [comment]
@pmorie

**Comment by eparis** [comment]
@erictune my thoughts are here (although disorganized as usual)  https://trello.com/c/1pn9IN3m/320-better-seccomp-support   But I think you nailed it.

**Comment by erictune** [comment]
I didn't realize that only increasingly more restrictive seccomp profiles can be loaded, if I am reading the comments on  https://trello.com/c/1pn9IN3m/320-better-seccomp-support  correctly.  That makes it much less flexible that I had imagined.

**Comment by eparis** [comment]
docker only lets us load a single profile, so that part isn't relevant today (I think RH is doing OCI work to make it a list of profiles but who knows if that ever gets plumbed up into docker)

If you load 2 profiles they will be executed in order and all of the profiles must 'approve' the syscall. If profile 1 says
`deny mount; allow all`
and profile 2 says
`deny module_init; allow mount; allow all`

Then BOTH `mount` and `module_init` will get denied.

**Comment by eparis** [comment]
So the profiles themselves do not have to be of increasing restrictivity (restrictivity is a word, right?) but the combination will always be at least as restrictive as either parent.

**Comment by jfrazelle** [comment]
What if instead of turning it off completely kubnetes loaded a custom default, w mount and whatever else you all need

**Comment by jfrazelle** [comment]
and if the user loads one it would override your default, just like how it works in docker.

**Comment by jfrazelle** [comment]
I'd be happy to contribute the patch depending on what syscalls you all need added

**Comment by vishh** [comment]
If users have a mechanism to override the default profile, I guess docker's default profile should itself work.

**Comment by jfrazelle** [comment]
So is the problem exposing the custom profile in the kubernetes API in a
time constraint? Just want to understand so I can help

On Monday, April 4, 2016, Vish Kannan <notifications@github.com> wrote:

> If users have a mechanism to override the default profile, I guess
> docker's default profile should itself work.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-205630885>
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by justincormack** [comment]
@erictune are you explicitly referring to Linux capabilities there? One option would be to tie the default Docker seccomp filter to the allowed capabilities so that adding say `CAP_SYS_ADMIN` allows some more syscalls, while `--cap-drop-all` removes a few more potentially.  In particular adding `CAP_SYS_ADMIN` would allow `unshare` and `mount`, allowing `systemd` to run.

Getting admins to write seccomp filters is somewhat unpleasant work, the aim with teh Docker seccomp support has been to avoid it as much as possible.

**Comment by thockin** [comment]
Is this manifesting as a real problem right now?  I am wild-guessing that it shows up as containers that fail (syscall blocked) with no remediation (since it didn't make it into our API) ?

It seems like an overly large thing to cherry pick, and yet not having it seems bad.  Can we maybe have an annotation that we pass through to docker, and add proper profiles and fields in 1.3?

**Comment by rhatdan** [comment]
I tried to add this pull request to docker, but it was closed because docker wants the tighter security

https://github.com/projectatomic/docker/pull/80

I believe that we should attempt to separate different security mechanisms from each other.  For example.  SELinux can block Capabilities, but I allow all Caps from an SELinux point of view and rely on the Capabilities Subsystem to manage caps,  Seccomp did not follow this standard, at least to a few syscalls that are controlled by SYS_ADMIN.  We saw these when trying to run systemd in a container.  
We are working on patches to allow the addition of syscalls to the white list as well as changing the default way that seccomp is enforced, so that users would be allowed to specify KILL versus ERRNO.  KILL Is the only seccomp action that outputs blocked syscalls to /var/log/audit/audit.log, which could give the user a clue on why he is getting permission denied.  

Allowing tools to specify Whitelist or blacklist syscalls would help.  It would also help if the syscall table that docker is using was provided externally to the tool, not compiled in. Then the admin or tools could just modify the default json and pass it to docker daemon on on the docker run command to specify the syscalls to be used by the container(s).

**Comment by jfrazelle** [comment]
If the default profile is the problem it could easily be fixed by my
suggestion of kubes passing its own w whatever other syscalls they need
added to the whitelist, fwict the problem is more time adding it to the API
unless I am mistaken

On Tuesday, April 5, 2016, Daniel J Walsh <notifications@github.com> wrote:

> I tried to add this pull request to docker, but it was closed because
> docker wants the tighter security
>
> projectatomic/docker#80 <https://github.com/projectatomic/docker/pull/80>
>
> I believe that we should attempt to separate different security mechanisms
> from each other. For example. SELinux can block Capabilities, but I allow
> all Caps from an SELinux point of view and rely on the Capabilities
> Subsystem to manage caps, Seccomp did not follow this standard, at least to
> a few syscalls that are controlled by SYS_ADMIN. We saw these when trying
> to run systemd in a container.
>
> We are working on patches to allow the addition of syscalls to the white
> list as well as changing the default way that seccomp is enforced, so that
> users would be allowed to specify KILL versus ERRNO. KILL Is the only
> seccomp action that outputs blocked syscalls to /var/log/audit/audit.log,
> which could give the user a clue on why he is getting permission denied.
>
> Allowing tools to specify Whitelist or blacklist syscalls would help. It
> would also help if the syscall table that docker is using was provided
> externally to the tool, not compiled in. Then the admin or tools could just
> modify the default json and pass it to docker daemon on on the docker run
> command to specify the syscalls to be used by the container(s).
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-205848108>
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by eparis** [comment]
In 1.2 we explicitly disable the seccomp profiles (aka we disable the docker default). So the 'manifestation' today is that kubernetes users cannot make use of this security feature provided by the underlying container backend.

The default docker seccomp profile would be great 99% of the time (the exact same % of the time that it would be great for docker). But until we plumb through some way to control the seccomp profile there wasn't a better alternative. This issue requires someone to plumb through the kube API a way to represent seccomp profiles and how to disable them.

I think there are a lot of ideas above how to plumb seccomp through the API, some form of which needs to be done before 1.3...

**Comment by bgrant0607** [comment]
We can't break existing applications running in existing clusters.

Users also expect to be able to delete and re-create their clusters across releases with identical behavior, and we have the official policy that we don't change default behavior of APIs without introducing a new API version.

**Comment by erictune** [comment]
@jfrazelle we don't know what calls our users might be doing, so we don't want to change the default and break them.   

I see that before 1.10 you [tested all the official docker images](https://github.com/docker/docker/pull/18780#issuecomment-167035672) and didn't see any issues.  Since then, have there been any user reports of images not working after upgrading to 1.10 due to seccomp?

**Comment by jfrazelle** [comment]
So maybe it could be something they could turn on to make it a
default? Like opt-in?

I completely understand the fear of breaking people on upgrade. I was
basically super stressed, on edge, nauseous (lol) the entire first few
weeks of our release since we shipped it as a default on upgrade. So I
hear you :)

I think opt in could help, also having large clusters or users who
have a lot of weird containers (this was me and @justincormack lol)
test profusely. If it gives you any warm fuzzies on the inside I will
say we tested A LOT, esp. things like mutli-arch. There are specific
syscalls for arm and 32bit architectures in the default whitelist.

I have some of the weirdest stuff in containers like chrome, skype
(which is 32bit running on 64bit) etc. Then apps like elasticsearch
where they create a seccomp profile in the app itself were all tested.
I created a database of all the syscalls used by official images and
tested them all. Of course nothing is perfect and I'm making no
promises but I was also super concerned about breaking people on
upgrade so I tried to do everything possible to at least put myself at
ease, which didn't really happen til a few weeks after. Phew. :)

On Tue, Apr 5, 2016 at 8:54 PM, Brian Grant <notifications@github.com> wrote:
> We can't break existing applications running in existing clusters.
>
> Users also expect to be able to delete and re-create their clusters across
> releases with identical behavior, and we have the official policy that we
> don't change default behavior of APIs without introducing a new API version.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu

**Comment by jfrazelle** [comment]
The only report as far as I can remember was the one linked here wrt adding
caps and having it still blocked by seccomp, we tried to clear it up with
the ability to fork our default and add things you needed like mount or
obviously run as "unconfined" I think that was where the most confusion
was, which in my opinion was minimal considering I was expecting the world
to break.

On Tue, Apr 5, 2016 at 9:07 PM, Eric Tune <notifications@github.com> wrote:

> @jfrazelle <https://github.com/jfrazelle> we don't know what calls our
> users might be doing, so we don't want to change the default and break
> them.
>
> I see that before 1.10 you tested all the official docker images
> <https://github.com/docker/docker/pull/18780#issuecomment-167035672> and
> didn't see any issues. Since then, have there been any user reports of
> images not working after upgrading to 1.10 due to seccomp?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-206110110>
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by pmorie** [comment]
@jfrazelle 

> So is the problem exposing the custom profile in the kubernetes API in a time constraint?

The major Q to me is how we should model this in the API; there are many facets that need to be thought through.  We want to avoid coupling the API to docker -- do we adopt OCI model of seccomp (https://github.com/opencontainers/runc/blob/519529febe2e0acb14e39cf54112c292ccb2eabe/libcontainer/configs/config.go#L28)?  Invent our own?  Stuff some opaque profile artifact into a ConfigMap?  Are seccomp profiles (either by definition or by reference) part of the pod-level security context, or the container-level security context?

Since we are at the very beginning of the 1.3 dev cycle I don't think time is an especially big concern at the moment (my own subjective opinion).

@eparis @jfrazelle Is there a format for seccomp profiles that is remotely container-runtime agnostic?

**Comment by jfrazelle** [comment]
As far as a realistic user friendly profile (that hopefully won't trap you
in a specific runtime) I'd have to suggest OCI

Or even using the OCI struct
https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L352
and
duplicating it into your own types (just incase they change it in the
future) which is what we did.

But for fun and putting it all out there, there is the "minijail policy"
for chromium
https://www.chromium.org/chromium-os/developer-guide/chromium-os-sandboxing#h.l7ou90opzirq

In my opinion (which I'm not super familiar with pods so take that w a
grain of salt) it should be per container context. Mostly because it should
be handled per process, so that if a app only needs to listen on a socket
they could ideally have a super restricted profile while it's database had
a less restrict one. Probably not the best example but yeah.



On Tuesday, April 5, 2016, Paul Morie <notifications@github.com> wrote:

> @jfrazelle <https://github.com/jfrazelle>
>
> So is the problem exposing the custom profile in the kubernetes API in a
> time constraint?
>
> The major Q to me is how we should model this in the API; there are many
> facets that need to be thought through. We want to avoid coupling the API
> to docker -- do we adopt OCI model of seccomp (
> https://github.com/opencontainers/runc/blob/519529febe2e0acb14e39cf54112c292ccb2eabe/libcontainer/configs/config.go#L28)?
> Invent our own? Stuff some opaque profile artifact into a ConfigMap? Are
> seccomp profiles (either by definition or by reference) part of the
> pod-level security context, or the container-level security context?
>
> Since we are at the very beginning of the 1.3 dev cycle I don't think time
> is an especially big concern at the moment (my own subjective opinion).
>
> @eparis <https://github.com/eparis> @jfrazelle
> <https://github.com/jfrazelle> Is there a format for seccomp profiles
> that is remotely container-runtime agnostic?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-206128224>
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by jfrazelle** [comment]
If we are being completely honest with ourselves though, no one (well
nearly no one) will be writing custom seccomp profiles. It is not something
that can be automated perfectly without the possibility of missing a
syscall, which was why the default profile was so important.


On Tuesday, April 5, 2016, Jessica Frazelle <me@jessfraz.com> wrote:

> As far as a realistic user friendly profile (that hopefully won't trap you
> in a specific runtime) I'd have to suggest OCI
>
> Or even using the OCI struct
> https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L352 and
> duplicating it into your own types (just incase they change it in the
> future) which is what we did.
>
> But for fun and putting it all out there, there is the "minijail policy"
> for chromium
> https://www.chromium.org/chromium-os/developer-guide/chromium-os-sandboxing#h.l7ou90opzirq
>
> In my opinion (which I'm not super familiar with pods so take that w a
> grain of salt) it should be per container context. Mostly because it should
> be handled per process, so that if a app only needs to listen on a socket
> they could ideally have a super restricted profile while it's database had
> a less restrict one. Probably not the best example but yeah.
>
>
>
> On Tuesday, April 5, 2016, Paul Morie <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> @jfrazelle <https://github.com/jfrazelle>
>>
>> So is the problem exposing the custom profile in the kubernetes API in a
>> time constraint?
>>
>> The major Q to me is how we should model this in the API; there are many
>> facets that need to be thought through. We want to avoid coupling the API
>> to docker -- do we adopt OCI model of seccomp (
>> https://github.com/opencontainers/runc/blob/519529febe2e0acb14e39cf54112c292ccb2eabe/libcontainer/configs/config.go#L28)?
>> Invent our own? Stuff some opaque profile artifact into a ConfigMap? Are
>> seccomp profiles (either by definition or by reference) part of the
>> pod-level security context, or the container-level security context?
>>
>> Since we are at the very beginning of the 1.3 dev cycle I don't think
>> time is an especially big concern at the moment (my own subjective opinion).
>>
>> @eparis <https://github.com/eparis> @jfrazelle
>> <https://github.com/jfrazelle> Is there a format for seccomp profiles
>> that is remotely container-runtime agnostic?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-206128224>
>>
>
>
> --
>
>
> Jessie Frazelle
> 4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
> pgp.mit.edu
> <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>
>
>

-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by pmorie** [comment]
@jfrazelle 

> As far as a realistic user friendly profile (that hopefully won't trap you in a specific runtime) I'd have to suggest OCI

My perception is that this is probably the thing we should model, yeah.

>  In my opinion (which I'm not super familiar with pods so take that w a grain of salt) it should be per container context.

The line of reasoning you presented here makes sense.

> If we are being completely honest with ourselves though, no one (well nearly no one) will be writing custom seccomp profiles. It is not something that can be automated perfectly without the possibility of missing a syscall, which was why the default profile was so important.

My own subjective opinion is that probably users will bucket into two groups:

1.  What's a seccomp profile?
2.  I already have it written

Where (2) is probably the minority of people w/ relatively stable applications that are already using seccomp to some degree.

So, here's the main question I am asking myself -- is there more utility to being able to fully define a seccomp profile directly in a container's security context, or is it better to reference them by name?
From a policy standpoint, it's probably easier to reference by name, and put policy on who can reference which name, rather than make a policy on who can do what to which syscall (which we would be effectively forced to do in openshift if we integrate a rich model of seccomp profile into the security context directly).  Any thoughts here @pweil-?

**Comment by pweil-** [comment]
> Any thoughts here @pweil-

Given the complexity of the model I think referencing by name is much easier than embedding it all within a security policy itself.  The policy may define a seccomp strategy that provides a default and may or may not allow the name to be overridden.  The permission to create custom seccomp profiles gates the ability to customize specific settings (like any other api resource).  The settings of the policy (and the permission to use said policy) gate your ability to override the usage of a profile.

**Comment by pmorie** [comment]
@pweil- sounds like you would be in favor of a seccomp profile resource that can be referenced in a container security context, with policy controlling:

1.  Who can create/update profiles
2.  Who can reference which profiles by name

I think if we have a profile resource, that resource should have a rich model of syscall/action etc like OCI, and in what I articulated above, I don't think a fine-grained policy for who can change the knobs on profile for which syscall would be needed.

Agree? Disagree?

**Comment by pweil-** [comment]
Yes, agree.

**Comment by justincormack** [comment]
I think people who have already written a profile for their specific application should in general apply it within the container as part of application setup on top of the default policy set by the cluster admin. The cluster policy should be for overall policies, and does not seem the right place to lock down a single application further as that needs to be in source control with the application as it will change over time.

**Comment by eparis** [comment]
@justincormack I generally agree with that. Or I think it should be part of the container definition and the container engine should load a list of policies. I'm ok if there is no way to load a custom defined policy per container in the first iteration. But given the difference in trust between the operator (who sets things in kube) and the container author/image I do see where some might want that feature 'someday'

**Comment by pmorie** [comment]
@justincormack I was being glib in my comment and should have said 'I know all about seccomp and know exactly what I want' instead of 'I already have it written'.  I think you raise a good point, though.  There are really probably 4 buckets of users:

1.  I don't know anything about seccomp
2.  I have heard of it, but don't know much about it
3.  I know about it and my application already uses libseccomp
4.  I know about it and I know what I want to do with the knobs, but my application doesn't use libseccomp currently

I think your case is (3).  I think we should allow (3), and facilitate (4) by adding an API for seccomp so that you don't have to make your application link against libseccomp.

**Comment by rhatdan** [comment]
I would say the two most common use cases are going to be.  

1. My app does not work with the default seccomp filters, how do I add the missing syscalls to whitelist?
2. I heard that this syscall is dangerous and my app does not use it, how do I remove it from whitelist?
3. How do I figure out which syscall is blocking my app from running?

**Comment by jfrazelle** [comment]
Agreed with @justincormack and @rhatdan, definitely no. 3 needs some
tooling for debugging, right now I will say it is not the most user
friendly experience. For no. 2 if it is really that bad we should remove it
from the default, unless its something like `open` ;)

On Thu, Apr 7, 2016 at 11:08 AM, Daniel J Walsh <notifications@github.com>
wrote:

> I would say the two most common use cases are going to be.
>
>    1. My app does not work with the default seccomp filters, how do I add
>    the missing syscalls to whitelist?
>    2. I heard that this syscall is dangerous and my app does not use it,
>    how do I remove it from whitelist?
>    3. How do I figure out which syscall is blocking my app from running?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-207031744>
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by jfrazelle** [comment]
What does the current pods security context look like vs. container
security context. You can just give me links to wherever it lives,
don't worry :) I'll try to find it too.

On Thu, Apr 7, 2016 at 12:14 PM, Jessica Frazelle <me@jessfraz.com> wrote:
> Agreed with @justincormack and @rhatdan, definitely no. 3 needs some tooling
> for debugging, right now I will say it is not the most user friendly
> experience. For no. 2 if it is really that bad we should remove it from the
> default, unless its something like `open` ;)
>
> On Thu, Apr 7, 2016 at 11:08 AM, Daniel J Walsh <notifications@github.com>
> wrote:
>>
>> I would say the two most common use cases are going to be.
>>
>> My app does not work with the default seccomp filters, how do I add the
>> missing syscalls to whitelist?
>> I heard that this syscall is dangerous and my app does not use it, how do
>> I remove it from whitelist?
>> How do I figure out which syscall is blocking my app from running?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>
>
>
>
> --
>
>
> Jessie Frazelle
> 4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
> pgp.mit.edu



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu

**Comment by thockin** [comment]
Any better conclusions here?  I hate disabling it, but without an escape
hatch it's bound to break someone.  And it is probably too late to enable
an escape hatch as a point release under 1.2 :(

**Comment by ncdc** [comment]
I think we're probably going to have to defer a real solution until 1.3. I know @pmorie is planning on writing up a proposal.

**Comment by jfrazelle** [comment]
I think it would be a good idea to not include it in say a .1 or patch
release because it is kinda changing behavior and is better for a .0, if
feel bad breaking semver

On Wednesday, April 20, 2016, Andy Goldstein <notifications@github.com>
wrote:

> I think we're probably going to have to defer a real solution until 1.3. I
> know @pmorie <https://github.com/pmorie> is planning on writing up a
> proposal.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/issues/20870#issuecomment-212433596>
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by thockin** [comment]
You *could* argue it was an oversight, but it's a stretch.  I was hoping
someone would convince me, but ...

**Comment by jfrazelle** [comment]
don't worry I'll be the pusher for 1.3 :innocent:

**Comment by thockin** [comment]
I'mma hold you to that.

**Comment by pmorie** [comment]
@jfrazelle 

> don't worry I'll be the pusher for 1.3

:+1:

**Comment by dims** [comment]
@thockin @vishh We can close this now (Since PR has merged https://github.com/kubernetes/kubernetes/commit/aa8c72adaaab6cba51ff868516ac5b423a95f069)

**Comment by ncdc** [comment]
Not yet - we need #26710 to merge first

**Comment by dchen1107** [comment]
Moved this to v1.3 for the proper feature tracking.

**Comment by bgrant0607** [comment]
Feature issue: https://github.com/kubernetes/features/issues/135

**Comment by tallclair** [comment]
Closing in favor of https://github.com/kubernetes/features/issues/135
